projects/clr/hipamd/src/hip_memory.cpp

```diff
@@ -172,15 +172,35 @@ hipError_t hipExternalMemoryGetMappedBuffer(void** devPtr, hipExternalMemory_t e
   if (devPtr == nullptr || extMem == nullptr || bufferDesc == nullptr || bufferDesc->flags != 0) {
     HIP_RETURN(hipErrorInvalidValue);
   }
+
   auto buf = reinterpret_cast<amd::ExternalBuffer*>(extMem);
-  const device::Memory* devMem = buf->getDeviceMemory(*hip::getCurrentDevice()->devices()[0]);
 
-  if (devMem == nullptr || ((bufferDesc->offset + bufferDesc->size) > devMem->size())) {
+  // Validate bounds
+  if (bufferDesc->size > buf->getSize() ||
```

Comment:
shouldn't this check just be "bufferDesc->size + bufferDesc->offset > buf->getSize()" ?